compiler/rustc_privacy/src/lib.rs

```diff
@@ -684,75 +681,26 @@ impl<'tcx> Visitor<'tcx> for EmbargoVisitor<'tcx> {
                     &self.effective_visibilities,
                 );
 
-                self.update_eff_vis(item.owner_id.def_id, impl_ev, None, Level::Direct);
+                if let Some(impl_ev) = impl_ev {
+                    self.update_eff_vis(item.owner_id.def_id, impl_ev, None, Level::Direct);
+                }
                 impl_ev
             }
             _ => self.get(item.owner_id.def_id),
         };
```

Comment:
Can this match be merged into the following one too?

Reply:
Yes, looks like that.